### ISSUE TITLE
[Attribute] Can't save checkbox attributes if unchecked

### ISSUE THREAD

**Issue by peteward** [Sylius/Sylius#4295]
https://github.com/Sylius/Sylius/blob/2b3d386df2ad60de968e19fb63720194083fa3fe/src/Sylius/Bundle/AttributeBundle/Form/EventSubscriber/BuildAttributeValueFormSubscriber.php

line 71 - the check on `isset($attributeValue['value'])` doesn't work for checkboxes since they don't submit anything with the page if unchecked and the exception below is thrown:

```
    /**
     * @param FormEvent $event
     */
    public function preSubmit(FormEvent $event)
    {
        $attributeValue = $event->getData();
        if (!isset($attributeValue['value']) || !isset($attributeValue['attribute'])) {
            throw new \InvalidArgumentException('Cannot create an attribute value form on pre submit event without "attribute" and "value" keys in data.');
        }
        $form = $event->getForm();
        $attribute = $this->attributeRepository->find($attributeValue['attribute']);
        $this->addValueField($form, $attribute);
    }
```

Removing this check works, just wondering if it is really needed or could/should be handled elsewhere - any idea @Zales0123 ?

**Comment by peteward** [comment]
It's a shame that the PR template doesn't quite work for auto-closing of issues, something to be looked at?

**Comment by michalmarcinkowski** [comment]
I think you need to specify the issue number at PR creation and it should auto-close it. When you add reference to the issue by editing already opened PR it will not work. That's what I observed :)

**Comment by peteward** [comment]
if you put it in the description with `fixes #xxxx` it closes - we do that locally and it works.  

It's probably the table format of the PR template that breaks it. From https://help.github.com/articles/closing-issues-via-commit-messages/

```
Closing multiple issues

To close multiple issues, preface each issue reference with one of the above keywords.

For example, This closes #34, closes #23, and closes example_user/example_repo#42 would close issues #34 and #23 in the same repository, and issue #42 in the "example_user/example_repo" repository.
```

I'll do a PR...

**Comment by michalmarcinkowski** [comment]
Thanks for the info :+1:
